devel/views.py

```diff
@@ -107,8 +107,9 @@ def stats(request):
 
 @login_required
 def clock(request):
-    devs = User.objects.filter(is_active=True).order_by(
-        'first_name', 'last_name').select_related('userprofile')
+    devs = User.objects.filter(is_active=True).filter(groups__in=Group.objects.filter(
+        Q(name='Developers') | Q(name='Trusted Users') | Q(name='Support Staff'))).order_by(
+        'first_name', 'last_name').select_related('userprofile').distinct()
 
```

Comment:
Instead of using 3 times Q I rather I will filter them with a python list comprehension

```
SELECTED_GROUPS = ['Developers', 'Trusted Users', 'Support Staff']


    def clock(request):
        groups = Group.objects.all().only('id', 'name')
        groups_ids = [x.id for x in groups if x.name in SELECTED_GROUPS]
        users = User.objects.filter(is_active=True).filter(
            groups__in=groups_ids).select_related('userprofile').distinct()
```

I'd go with doing an implementation like this ^ for two reasons, despite if it's optimized or not (I don't think either this version is actually a super optimized version):

1.- Legibility (Reading between all Q queries mixed with the rest of the queries might lead to confusion), and a raw query will be even worst for that.
2.- Maintainability (just adding/removing another value to the list SELECTED_GROUP is easier than adding a Q condition with the high chances of screwing it everything)